builder/simplify.cc

```diff
@@ -517,9 +517,6 @@ class simplifier : public node_mutator {
     for (std::size_t d = 0; d < op->dims.size(); ++d) {
       interval_expr bounds_d = mutate(op->dims[d].bounds);
       dim_expr new_dim = {bounds_d, mutate(op->dims[d].stride), mutate(op->dims[d].fold_factor)};
-      if (!is_zero(new_dim.stride) && prove_true(new_dim.fold_factor == 1 || new_dim.bounds.extent() == 1)) {
```

Comment:
Why were we doing this in the first place?

Reply:
At one point, this was *replacing* fold factor 1 with stride 0. Folding has/had some limitations, so this worked in a case where folding didn't. I think that's been long fixed, and we stopped dropping the fold factor anyways. So it's mostly a historical workaround I guess.